### PR TITLE
#82 Improve godoc

### LINKED ISSUE #82: Improve Go documentation
Go comments are missing for most public API.
To simplify working with the Go driver we should document all public API so that it is available on [the Go reference page](https://pkg.go.dev/github.com/exasol/exasol-driver-go#section-documentation).

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #82

### CODE REVIEW

File: driver.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// Package exasol implements a driver for the Exasol database.
```
Comment (Nightapes) [inline driver.go:1]: Should we also add a copyright here or add usage example like https://github.com/go-sql-driver/mysql/blob/master/driver.go mysql is doing it?

Comment (kaklakariada) [reply driver.go:6]: 👍 ✅

File: doc/changes/changes_0.4.7.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# exasol-driver-go 0.4.7, released 2022-10-??
+
+Code name: Improve documentation
```
Comment (ckunki) [inline doc/changes/changes_0.4.7.md:3]:
```suggestion
Code name: Improve documentation and make internal APIs private.
```

File: doc/changes/changes_0.4.7.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+## Summary
+
+In this release we improved the Godoc documentation available at [pkg.go.dev](https://pkg.go.dev/github.com/exasol/exasol-driver-go). We also made internal types private that are used for the communication with the database.
+
```
Comment (ckunki) [inline doc/changes/changes_0.4.7.md:8]:
```suggestion

Additionally this release makes some internal APIs private. Affected type are `connection`, `command`, `baseResponse`.
```

**Review by ckunki — CHANGES_REQUESTED** [review]

**Review by ckunki — APPROVED** [review]

**Pull request merged**